pkg/asset/agent/image/agentimage.go

```diff
@@ -131,6 +126,44 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 	return nil
 }
 
+// updateIgnitionInfo updates igninfo.json with the current size of
+// the ignition buffer.
+func (a *AgentImage) updateIgnitionInfo(ignitionSize int) error {
+	ignInfoPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
+	ignInfoRaw, err := os.ReadFile(ignInfoPath)
+	if err != nil {
+		return err
+	}
+
+	type ignInfo struct {
+		File   string `json:"file"`
+		Length int64  `json:"length"`
+		Offset int64  `json:"offset"`
+	}
+	var info ignInfo
+	err = json.Unmarshal(ignInfoRaw, &info)
+	if err != nil {
+		return err
+	}
+
+	info.Length = int64(ignitionSize)
```

Comment:
We must not shorten the size of the ignition image. This would break the user workflow where they use coreos-installer to extract the ignition, modify it, and then reinsert it. We must continue to pad the image to at least its original size and indicate that in the igninfo.json.

Reply:
Unfortunately that was a pre-existing issue (lack of padding), I'll add a fix in `updateIgnitionImg`.
In that way the stored length information would be always the max between 256Kb and the actual compressed cpio archive length.

Reply:
> Unfortunately that was a pre-existing issue (lack of padding)

I don't think it was, because we don't pass `O_TRUNC`.
I really hope it wasn't.

Reply:
> In that way the stored length information would be always the max between 256Kb and the actual compressed cpio archive length.

I think a good formula would be something like e.g.:

```
(1 + ((ignitionSize + (ignitionEmbedAreaSize / 4)) / ignitionEmbedAreaSize) * ignitionEmbedAreaSize
```

so it would grow in increments of 256kiB, and always have at least 32kiB free.

Reply:
> > In that way the stored length information would be always the max between 256Kb and the actual compressed cpio archive length.
> 
> I think a good formula would be something like e.g.:
> 
> ```
> (1 + ((ignitionSize + (ignitionEmbedAreaSize / 4)) / ignitionEmbedAreaSize) * ignitionEmbedAreaSize
> ```
> 
> so it would grow in increments of 256kiB, and always have at least 32kiB free.

On a second thought, given that the `coreos-installer` logic uses the reported (or actual) file size, probably we don't need that?

Reply:
> On a second thought, given that the coreos-installer logic uses the reported (or actual) file size, probably we don't need that?

I meant the ideally we should pad to that size, so that the user would always have some space left to edit the ignition if they needed to, even if they started with huge manifests.

Reply:
wouldn't this also be covered by regenerating the ISO with the newly updated manifests (ensuring that the new size will always be addressed)? If we are ok I'd would land the current implementation, so that we could address the initial issue.

Reply:
Right now editing the ignition works, and after this patch huge manifests will work. But both at the same time won't, because if the manifests are huge we don't leave any empty space.
So even without adding extra padding this is strictly better than what we had, but if we backport only this then there's still a risk that there'll be more complaints in future.